edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradebookFrameworkService.java

```diff
@@ -79,4 +80,31 @@
 	 */
 	public void setDefaultGradingScale(String uid);
 
+	/**
+	 *
+	 *	returns all the Available Grading Scales in the system.
+	 */
+	public List getAvailableGradingScales();
+
+	/**
+	 * Adds a new grade scale to an existing gradebook.
+	 *
+	 * @param scaleUuid
+	 *   The Uuid of the scale we want to be added to the gradebook
+	 * @param gradebook
+	 *   The gradebook with GradeMappings where we will add the grading scale.
+	 *
+	 */
+	public void saveGradeMappingToGradebook(String scaleUuid, Object gradebook);
+
+
+	/**
+	 * Update a grademapping with new values.
+	 *
+	 * @param gradeMapping
+	 *   The gradeMApping to update.
+	 *
+	 */
+	public void updateGradeMapping(Object gradeMapping);
```

Comment:
Yes, I agree that passing the object is not a great thing, but GradeMapping is not defined in the API. The API is a dependence, so if I try to import these things from the hibernate implementation, it, logically, generates a cyclic reference. The only way I've found is to pass as object and later in the implementation convert it to the right type.   

It is done in this other method in the same place:
/**
     * @param gradingScaleDefinitions
     *	A collection of GradingScaleDefinition beans.
	 */
	public void setAvailableGradingScales(Collection gradingScaleDefinitions);

It passed a Collection of gradinScaleDefinitions but they are objects... not really gradinScaleDefinitions... because this is not in the API.

I'm open to any suggestion to change this but of course, add this to the API is a big work, becuase GradeMappindg has a Gradebook and a GradingScale "inside" so I would need to add that to the API too... and surely other things after that in cascade...